### ISSUE TITLE
Consider global translation strings handling in UI Library components

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#4639]
Currently, each UI component accepts a prop, `i18n`, which contains all of the strings needed for that component. This helps keep locale strings connected to their component, and so when a component is removed the strings are no longer loaded on the frontend.

However, there are a couple of issues with this:

- Lots of string duplication in the client. Not a major performance issue but unnecessary.
- Each `i18n` object stores the strings with a local key, so there's a little bit of hunting to find out what the original locale key is.
- The global Vue.js mixin that merges params with locale strings only works on the `i18n` object, so if we want to pass a locale string in some other way we have to do the merging again.

The refactor for #2906 implemented a global way of passing constants to the frontend. In a component's PHP configuration object, a constant can be passed like this:

```php
$templateMgr = \TemplateManager::getManager($request);
$templateMgr->setConstant('WORKFLOW_STAGE_ID_SUBMISSION');
```

This makes the constant globally available in the client (`pkp.const`), but also makes sure that only constants requested by a component that exists in the UI are passed there. When the component is refactored to remove the constant dependency, it will be removed from the UI as well.

I'd like to do something similar for locale strings. It should be something like:

```php
$templateMgr = \TemplateManager::getManager($request);
$templateMgr->setI18n('locale.string.key');
```

If we want to merge params before it is sent to the UI, we can use:

```php
$templateMgr = \TemplateManager::getManager($request);
$templateMgr->setI18n('locale.string.key', ['param' => $value]);
```

In Vue.js components, there will no longer be a `i18n` property. Instead, we should match the `__()` usage on the backend:

```html
<div class="title">
  {{ __('locale.string.key', {param: value}) }}
</div>
```

This may mean that we lose out on the string documentation in the UI Library itself, but in return we get an easy way to always find a localized string in the UI Library codebase (search for `__(`).

It would be better to do this sooner rather than later. Each new component we add will mean more refactoring later.

**Comment by NateWr** [comment]
One thing I just thought of... in the following case:

```php
$templateMgr->setI18n('locale.string.key', ['param' => $value]);
```

We may encounter a situation where we want to use `locale.string.key` twice, but with different params:

```php
$templateMgr->setI18n('locale.string.key', ['param' => $valueOne]);
$templateMgr->setI18n('locale.string.key', ['param' => $valueTwo]);
```

In such cases, the `locale.string.key` would be overwritten with `$valueTwo`. Whenever possible, we should prefer to merge the params in the browser rather than the server. However, when that is not an option, perhaps we can use a common syntax, like `-<identifier>`:

```php
$templateMgr->setI18n('locale.string.key-valueOne', ['param' => $valueOne]);
$templateMgr->setI18n('locale.string.key-valueTwo', ['param' => $valueTwo]);
```

**Comment by NateWr** [comment]
Implemented in #5865.
